opencollective/opencollective-api
# PayPal Payout workers

I was able to test this using PayPal's sandbox and it was a bit annoying, feel free to review the code and I'll make sure to set this up in staging once it is merged.

If you really want to test locally:
1. Enable payouts in your host updating settings to have `features.paypalPayouts = true`.
2. Go to your host settings -> sending money and use the Application from https://developer.paypal.com/developer/applications/
    - Use _PayPal OC Inc_ credentials from 1Password
3. Create an expense using PayPal email `paypal-buyer@opencollective.com` to any of the collectives hosted by the host you just connected
4. Approve and Schedule the expense for Payment
5. Manually run the workers
    1. `npm run script cron/hourly/10-pay-paypal-scheduled-expenses-payouts.js` should mark the expense as _processing_
    2. `npm run script cron/hourly/11-check-pending-paypal-expenses.js` should mark the expense as _paid_
